### PR TITLE
chore: enable pre-commit and fix some issues

### PR DESCRIPTION
**Pull request by trsergey25**

### CODE REVIEW

**Review by VladUsenko26 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -46,7 +46,9 @@
     "eslint-config-prettier": "9.1.0",
     "eslint-plugin-jest": "27.9.0",
     "eslint-plugin-prettier": "5.1.3",
+    "husky": "^9.0.11",
```
Comment (RomanBobrovskiy) [inline package.json:49]: let's use strict versions

File: package.json [hunk 2]
```diff
@@ -66,5 +68,8 @@
     "packages/tsconfig",
     "examples/app-router",
     "examples/page-router"
-  ]
+  ],
+  "lint-staged": {
+    "*.{ts,tsx}": "eslint"
```
Comment (RomanBobrovskiy) [inline package.json:73]:
```suggestion
    "*.{ts,tsx}": "eslint --fix"
```

File: package.json (hunk 1 above)
Comment (trsergey25) [reply package.json]: okey

File: package.json (hunk 2 above)
Comment (trsergey25) [reply package.json]: done

**Review by RomanBobrovskiy — APPROVED** [review]

**Pull request merged**